### PR TITLE
Add ASAP SKU support.

### PR DESCRIPTION
**Pull request by vimish**

### CODE REVIEW

File: VMEncryption/main/DiskUtil.py [hunk 1]
```diff
@@ -887,10 +890,15 @@ def get_azure_data_disk_controller_and_lun_numbers(self, dev_items_real_paths):
         """
         Return the controller ids and lun numbers for data disks that show up in the dev_items
         """
+        list_devices = []
+        is_os_nvme, root_device_path_nvme = self.is_os_disk_nvme()
 
+        if is_os_nvme:
```
Comment (pankajosh) [inline VMEncryption/main/DiskUtil.py:896]: Is it true to say that if OS disk is NVMe then other all disks must be NVMe.

Comment (vimish) [reply VMEncryption/main/DiskUtil.py:896]: Yes. For the new ASAP SKU's both OS and Data disk will be NVMe. Resource and BEK disk are SCSI.

File: VMEncryption/main/EncryptionSettingsUtil.py [hunk 2]
```diff
@@ -247,10 +247,16 @@ def get_settings_data(self, protector_name, kv_url, kv_id, kek_url, kek_kv_id, k
...
+        is_os_nvme, root_device_path_nvme = disk_util.is_os_disk_nvme()
         if os.path.exists(root_device_path):
             root_device_path = os.path.realpath(root_device_path)
+        elif os.path.exists(root_device_path_scsi):
+            root_device_path = os.path.realpath(root_device_path_scsi)
+        elif is_os_nvme and os.path.exists(root_device_path_nvme):
+            root_device_path = os.path.realpath(root_device_path_nvme)
         else:
-            root_device_path = os.path.realpath(os.path.join(CommonVariables.azure_symlinks_dir, "scsi0/lun0"))
+            self.logger.log("Cannot locate root device")
```
Comment (bamuir) [inline VMEncryption/main/EncryptionSettingsUtil.py:259]: If the root device is not found, and we are no longer defaulting to "root_device_path = os.path.realpath(os.path.join(CommonVariables.azure_symlinks_dir, "scsi0/lun0"))", how does the error come across on line 260 if root_device_path is None? 

I can't think of how to test this right now, but it seems there might be a TypeError when root device is not found. Wondering if we want to throw a more specific error ?

**Review by bamuir — APPROVED** [review]
Looks good to me. Would we be able to measure adoption / usage currently using the VM size on our existing tables?

File: VMEncryption/main/DiskUtil.py [hunk 3]
```diff
@@ -1203,6 +1291,18 @@ def get_azure_devices(self):
...
+    
+    def get_azure_nvme_os_devices(self):
+        devices = []
+        is_os_nvme, os_block_device = self.is_os_disk_nvme()
+        if is_os_nvme:
+            proc_comm = ProcessCommunicator()
+            self.command_executor.ExecuteInBash('ls {0}*'.format(os_block_device), True, communicator=proc_comm)
+            os_devices = re.split('[ \n]', proc_comm.stdout)
+            for os_device in os_devices:
+                if os_device.startswith(os_block_device) and os.path.exists(os_device):
```
Comment (Jyotsna-Anand) [inline VMEncryption/main/DiskUtil.py:1303]: Is there a case when os_device can be null? Should there be a check to ensure os_device is not null, before invoking these other methods on os_device?

**Review by Jyotsna-Anand — APPROVED** [review]

File: VMEncryption/main/EncryptionSettingsUtil.py (hunk 2 above)
Comment (vimish) [reply VMEncryption/main/EncryptionSettingsUtil.py:259]: The default to scsi0/lun0 was breaking in NVMe VM's as it did not exist. An error was thrown when trying to stamp the disks. I believe the same error will be thrown if we are not able to locate the root device.

But, I get the point. I will look into a better error handling probably in the next PR.

File: VMEncryption/main/DiskUtil.py (hunk 3 above)
Comment (vimish) [reply VMEncryption/main/DiskUtil.py:1303]: If OS is determined to be NVMe then we will always have an OS device.
Also, I am planning to do a review of error handling for this new path. I will add more checks then.

**Pull request merged**